packages/dynamite/dynamite_runtime/lib/src/utils/json_schema_validation.dart

```diff
@@ -0,0 +1,96 @@
+import 'package:built_collection/built_collection.dart';
+import 'package:built_value/json_object.dart';
+
+/// Checks the [input] against the given String validators.
+///
+/// Throws an `FormatException` when the input is not:
+/// * matching the given [pattern].
+/// * longer than [minLength].
+/// * shorter than [maxLength].
+void checkString(
+  dynamic input,
+  String parameterName, {
+  int? maxLength,
+  int? minLength,
+  RegExp? pattern,
+}) {
+  String value;
+  if (input is JsonObject && input.isString) {
+    value = input.asString;
+  } else if (input is String) {
+    value = input;
+  } else {
+    return;
```

Comment:
Shouldn't this rather throw an argument error because the method was called with the wrong data?

Reply:
I don't think so; ArrayValidator.maxItems for example says:
>  An array instance is valid against "maxItems" if its size is less than, or equal to, the value of this keyword.

https://json-schema.org/draft/2020-12/draft-bhutton-json-schema-validation-01#section-6.4.1

To my understanding any non `Iterable` value is always valid. Dynamite shouldn't ever try to apply such a validation to a non `JsonObject`, `Iterable` like or `Null` object but why potentially break at runtime when it is not wrong to just return.